src/lib/loginHandler.ts

```diff
@@ -14,19 +14,23 @@
  *  limitations under the License.
  */
 
-import {ClientOptions} from './clientOptions'
-import OAuthAgentConfiguration from './oauthAgentConfiguration'
-import {generateHash, generateRandomString} from './pkce'
+import * as urlparse from 'url-parse'
+import {ClientOptions} from './clientOptions';
+import OAuthAgentConfiguration from './oauthAgentConfiguration';
+import {generateHash, generateRandomString} from './pkce';
+import {AuthorizationRequestData} from './authorizationRequestData';
+import {AuthorizationResponseException} from './exceptions'
+
+export async function createAuthorizationRequest(config: OAuthAgentConfiguration, options?: ClientOptions): Promise<AuthorizationRequestData> {
```

Comment:
Both methods in this handler do not need `async`, I think.

Reply:
fixed